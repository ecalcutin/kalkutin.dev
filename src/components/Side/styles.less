@import "theme/variables.less";

.aside {
  width: 40px;
  position: fixed;
  bottom: 0;
  color: @color-secondary-light;
}

.left {
  left: 40px;

  @media (max-width: 1080px) {
    left: 20px;
  }

  @media (max-width: 768px) {
    display: none;
  }
}
.right {
  right: 40px;

  @media (max-width: 1080px) {
    right: 20px;
  }

  @media (max-width: 768px) {
    display: none;
  }
}
