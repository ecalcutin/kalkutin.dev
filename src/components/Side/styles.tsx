import styled from "styled-components";

export const StyledSideElement = styled.div<{ orientation: "left" | "right" }>`
  width: 40px;
  position: fixed;
  bottom: 0;
  left: ${(props) => (props.orientation === "left" ? "40px" : "auto")};
  right: ${(props) => (props.orientation === "left" ? "auto" : "40px")};
  z-index: 10;
  color: var(--light-slate);

  @media (max-width: 1080px) {
    left: ${(props) => (props.orientation === "left" ? "20px" : "auto")};
    right: ${(props) => (props.orientation === "left" ? "auto" : "20px")};
  }

  @media (max-width: 768px) {
    display: none;
  }
`;
