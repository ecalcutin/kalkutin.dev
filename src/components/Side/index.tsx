import React from 'react';

import { StyledSideElement } from './styles';

type SideProps = {
  orientation: 'left' | 'right';
  children: React.ReactElement;
};

const Side = (props: SideProps) => {
  const { orientation, children } = props;

  return (
    <StyledSideElement orientation={orientation}>{children}</StyledSideElement>
  );
};

export default Side;
