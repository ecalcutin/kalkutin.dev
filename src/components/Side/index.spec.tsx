import renderer from "react-test-renderer";

import Side from "./index";

describe("Sidebar", () => {
  it("should render correctly on left", () => {
    const tree = renderer.create(<Side orientation="left">content</Side>);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("should render correctly on right", () => {
    const tree = renderer.create(<Side orientation="right">content</Side>);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
