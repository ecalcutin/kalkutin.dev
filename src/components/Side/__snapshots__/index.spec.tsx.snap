// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Side bar should render correctly 1`] = `
<div
  className="sc-bdnxRM bFZpkQ"
  orientation="left"
>
  content
</div>
`;

exports[`Side bar should render correctly on left 1`] = `
<div
  className="sc-bdnxRM bFZpkQ"
  orientation="left"
>
  content
</div>
`;

exports[`Side bar should render correctly on right 1`] = `
<div
  className="sc-bdnxRM kmaXNM"
  orientation="right"
>
  content
</div>
`;

exports[`Sidebar should render correctly on left 1`] = `
<div
  className="sc-bdnxRM bFZpkQ"
  orientation="left"
>
  content
</div>
`;

exports[`Sidebar should render correctly on right 1`] = `
<div
  className="sc-bdnxRM kmaXNM"
  orientation="right"
>
  content
</div>
`;
